dispatchers.go

```diff
@@ -313,6 +349,22 @@ func workdir(b *Builder, args []string, attributes map[string]bool, flagArgs []s
 	return nil
 }
 
+func parseCommandForHeredoc(firstline string) string {
+	// Expected format `RUN <str> <<`
+	if strings.HasPrefix(firstline, "RUN ") || strings.HasPrefix(firstline, "run ") {
+		if strings.HasPrefix(firstline, "RUN ") {
+			firstline = strings.TrimPrefix(firstline, "RUN ")
+		} else if strings.HasPrefix(firstline, "run ") {
+			firstline = strings.TrimPrefix(firstline, "run ")
+		}
+		if strings.Contains(firstline, "<<") {
+			arr := strings.Split(firstline, "<<")
+			return strings.TrimSpace(arr[0])
+		}
+	}
+	return ""
+}
```

Comment:
Are you guaranteed not to have a space in front of "run" or "RUN" at this point? I think so, but things can go awry here if not.

Reply:
Yes at this point this RUN instruction is already parsed, anything other than this will be rejected by the parser in earlier cycles.

Reply:
Added test cases for these in `dispatcher_test.go` in function `TestParseHeredocCommand`

Reply:
Are we testing syntax like this example from that section?
```Dockerfile
RUN <<FILE1 cat > file1 && <<FILE2 cat > file2
I am
first
FILE1
I am
second
FILE2
```

Reply:
Added test cases for this with changed code.